src/DiagnosticMessages.ts

```diff
@@ -758,6 +758,11 @@ export let DiagnosticMessages = {
         message: `'${symbol}' is incompatible across scope group (${scopeName})`, // TODO: Add scopes where it was defined
         code: 1145,
         severity: DiagnosticSeverity.Error
+    }),
+    nameCollision: (thisThingKind: string, thatThingKind: string, thatThingName: string) => ({
+        message: `${thisThingKind} has same name as ${thatThingKind} '${thatThingName}'`,
+        code: 1146,
+        severity: DiagnosticSeverity.Error
```

Comment:
could we delete `duplicateClassDeclaration`'s code and steal its diagnostic code? Do we still need that `duplicateClassDeclaration` diagnostic anymore?

Reply:
Sounds good ...